MiniApp/Classes/core/Models/MiniAppManifest.swift

```diff
@@ -0,0 +1,40 @@
+/// Mini App Meta data information
+internal struct MetaDataResponse: Decodable {
+    var reqPermissions: [MACustomPermissionsResponse]?
+    var optPermissions: [MACustomPermissionsResponse]?
+
+    var exampleHostAppMetaData: [String: String]?
+
+    private enum CodingKeys: String, CodingKey {
+        case reqPermissions,
+             optPermissions,
+             exampleHostAppMetaData
+    }
+}
+
+internal struct MACustomPermissionsResponse: Decodable {
+    var name: String?
+    var reason: String?
+
+    private enum CodingKeys: String, CodingKey {
+        case name,
+             reason
+    }
+}
+
+/// Mini-app meta data information
+public struct MiniAppManifest: Codable {
+
+    /// List of required permissions for a mini-app
+    public var requiredPermissions: [MASDKCustomPermissionModel]?
+    /// List of optional permissions for a mini-app
+    public var optionalPermissions: [MASDKCustomPermissionModel]?
+    /// Key-value pair data that is received from the endpoint
+    public var exampleHostAppMetaData: [String: String]?
```

Comment:
Sorry, I think I made this kind of confusing. `exampleHostAppMetaData` shouldn't be part of our SDK - it's just an example key. 

The Host App can use any additional keys that they want in the manifest. So we need a way to provide the entire manifest response to them. i.e. have a `manifest` variable here which maps the top-level key/values of the whole manifest. For example, the host app can call:

```
// This returns the raw string value: {"exampleKey": "test"}
// It's the Host App's responsibility to parse it
miniAppManifest.manifest["exampleHostAppMetaData"]
```

Reply:
Sorry. I just added the variable there with the mock json for understanding. Once the endpoints are ready, I will change something like `manifest` or `extras`. 

variable name defined here will be static but it will retrieve any dynamic key from the response.